server/models/db/workerTimeslot.js

```diff
@@ -21,7 +21,20 @@ async function updateWorkerAvailability(workerId, specificDate, newStatus) {
     return db.any(`update ${Tables.workerTimeslot} set status='${newStatus}' where date='${specificDate}' and worker_id='${workerId}';`);
 }
 
+async function checkWorkerAvailability(workerTimeslotId) {
+    //Selects all from workerTimeslot table 
+    const queryStatement = `select * from ${Tables.workerTimeslot}  where worker_timeslot_id = ${workerTimeslotId}`;
+    const queryOutput = await db.any(queryStatement);
+    //check if workertimeSlot is present 
+    if (_.isEmpty(queryOutput)) return false;
+    //check if available 
+    const currentStatus = _.includes(arr,'available');
```

Comment:
Hi Nancy,
`arr` is not defined. My recommendation was to keep line 32 and adjust the conditional check on line 33 to see if the `currentStatus` includes TimeslotStatus.available, e.g., `if (_.includes(currentStatus, TimeslotStatus.available)) return true;`

Reply:
Hi Amy,
I see! Thank you for your clarifying!
My Action: I have changed reverted the changes to line 31 and replaced  the check on line 32 with` if (_.includes(currentStatus, TimeslotStatus.available)) return true;`

You can see this change in line 32 of server/routes/private.js

Please confirm if this takes care of your suggestion!

Reply:
Hi Nancy, 
Yep, it is possible to just optimize this line to return `_.includes(currentStatus, TimeslotStatus.available)` for more concise code but this works as well.

Reply:
Hi Amy,
It seems there have been 2 conflicting suggestions for this line.
Suggestion 1: _.includes(currentStatus, TimeslotStatus.available)
Suggestion 2: recommend keeping this line as _.map(...).

Could you specify which one is the suggestion I should implement?

Thanks!
Nancy

Reply:
Hi Nancy,
I believe you have now fixed the problem as per the comments below so no further action is needed for this comment.